src/main/java/com/conveyal/taui/persistence/FilePersistence.java

```diff
@@ -0,0 +1,31 @@
+package com.conveyal.taui.persistence;
+
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.conveyal.taui.AnalysisServerConfig;
+import spark.Response;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+public abstract class FilePersistence {
+    public static final FilePersistence in;
+
+    static {
+        // Create a single central class for file persistence, either using S3 or using
```

Comment:
This comment looks incomplete.  It might also be worth noting where future classes for other file storage providers could be patched in.